Source/Renderer/FramebufferManager.js

```diff
@@ -0,0 +1,308 @@
+import DeveloperError from "../Core/DeveloperError.js";
+import PixelFormat from "../Core/PixelFormat.js";
+
+/**
+ * Creates a wrapper object around a framebuffer and its resources.
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {Number} [options.colorAttachmentsLength=1] The number of color attachments this FramebufferManager will create.
+ * @param {Boolean} [options.color=true] Whether the FramebufferManager will use color attachments.
+ * @param {Boolean} [options.depth=false] Whether the FramebufferManager will use depth attachments.
+ * @param {Boolean} [options.depthStencil=false] Whether the FramebufferManager will use depth-stencil attachments.
+ * @param {Boolean} [options.createColorAttachments=true] Whether the FramebufferManager will construct its own color attachments.
+ * @param {Boolean} [options.createDepthAttachments=true] Whether the FramebufferManager will construct its own depth attachments.*
+ *
+ * @exception {DeveloperError} Cannot have both a depth and depth-stencil attachment.
+ *
+ * @private
+ * @constructor
+ */
+function FramebufferManager(options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+  this._colorAttachmentsLength = defaultValue(
+    options.colorAttachmentsLength,
+    1
+  );
+
+  this._color = defaultValue(options.color, true);
+  this._depth = defaultValue(options.depth, false);
+  this._depthStencil = defaultValue(options.depthStencil, false);
+  //>>includeStart('debug', pragmas.debug);
+  if (this._depth && this._depthStencil) {
+    throw new DeveloperError(
+      "Cannot have both a depth and depth-stencil attachment."
+    );
+  }
+  //>>includeEnd('debug');
+
+  this._createColorAttachments = defaultValue(
+    options.createColorAttachments,
+    true
+  );
+  this._createDepthAttachments = defaultValue(
+    options.createDepthAttachments,
+    true
+  );
+
+  this._useHdr = false;
+
+  this._framebuffer = undefined;
+  this._colorRenderbuffer = undefined;
+  this._colorTextures = [];
+  this._depthStencilRenderbuffer = undefined;
+  this._depthStencilTexture = undefined;
+  this._depthRenderbuffer = undefined;
+  this._depthTexture = undefined;
+
+  this._attachmentsSet = false;
+}
+
+Object.defineProperties(FramebufferManager.prototype, {
+  framebuffer: {
+    get: function () {
+      return this._framebuffer;
+    },
+  },
+  status: {
+    get: function () {
+      return this._framebuffer.status;
+    },
+  },
+});
+
+FramebufferManager.prototype.isDirty = function (width, height, hdr) {
+  hdr = defaultValue(hdr, false);
+
+  var texturesDirty = false;
+  var length = this._colorTextures.length;
+  var texture;
+  for (var i = 0; i < length; ++i) {
+    texture = this._colorTextures[i];
+    if (
+      !defined(texture) ||
+      texture.width !== width ||
+      texture.height !== height
+    ) {
+      texturesDirty = true;
+      break;
+    }
+  }
+  return length === 0 || texturesDirty || this._useHdr !== hdr;
```

Comment:
This function does not consider whether `setAttachments` is true or false. I was unsure about this but I think it makes sense because of the logic in `update`.. 

Also, now that it's possible to not create color attachments, should this check depth attachments too? What about renderbuffers?

Reply:
Resolved by having `FramebufferManager` store its own width and height; `isDirty` checks those instead.